### PR TITLE
feat: add `TryWith` and `Do` to `Verifier`

### PR DESCRIPTION
**Pull request by felix-berg**
In the case where the return type of an effects operation is `Void`, `op.tpe` has type `MonoType.Native(java.lang.Object)`. However, this doesn't match with the associated type of `Do.Expr` expressions reffering to such operations. 
Is this expected? Looks like it has something to do with the way void is handled during the TypeInference-phase (now removed...). More details in TODO comment.

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/language/phase/Verifier.scala [hunk 1]
```diff
@@ -442,12 +443,50 @@ object Verifier {
...
+      val op = eff.ops.find(_.sym == opUse.sym)
+        .getOrElse(throw InternalCompilerException(s"Unknown operation sym: '${opUse.sym}'", loc))
+
+      val oprestype = op.tpe match {
+        case MonoType.Void => tpe // should match any return type
+        case t => t
+      }
+
+      val sig = MonoType.Arrow(ts, tpe)
+      val opsig = MonoType.Arrow(
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/Verifier.scala:485]: Deadcode? How are these locals used?

**Comment by magnus-madsen** [comment]
Ill let @JonathanStarup have a look. 

Perhaps the code style can be improved a bit, not sure.

File: main/src/ca/uwaterloo/flix/language/phase/Verifier.scala (hunk 1 above)
Comment (felix-berg) [reply main/src/ca/uwaterloo/flix/language/phase/Verifier.scala:485]: They are only used on line 489. Should they be removed and just inlined into the checkEq call?

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/Verifier.scala:485]: No, its my bad :)

File: main/src/ca/uwaterloo/flix/language/phase/Verifier.scala [hunk 2]
```diff
@@ -442,12 +443,50 @@ object Verifier {
...
 
-    case Expr.TryWith(_, _, _, _, tpe, _, _) =>
-      // TODO: VERIFIER: Add support for TryWith.
-      tpe
+    case Expr.TryWith(exp, effUse, rules, ct, tpe, purity, loc) =>
+      val exptype = visitExpr(exp) match {
+        case MonoType.Arrow(List(u), t) =>
+          check(expected = MonoType.Unit)(actual = u, loc)
+          t
+        case e => failMismatchedShape(e, "Arrow with one parameter", loc)
```
Comment (JonathanStarup) [inline main/src/ca/uwaterloo/flix/language/phase/Verifier.scala:451]: I think these errors needs to be normalized/refactored at some point but for now its fine

**Review by JonathanStarup — APPROVED** [review]

File: main/src/ca/uwaterloo/flix/language/phase/Verifier.scala (hunk 2 above)
Comment (felix-berg) [reply main/src/ca/uwaterloo/flix/language/phase/Verifier.scala:451]: yea, i'll do it in a seperate PR.

File: main/src/ca/uwaterloo/flix/language/phase/Verifier.scala [hunk 3]
```diff
@@ -442,12 +443,50 @@ object Verifier {
       val t = visitExpr(exp)
       checkEq(tpe, t, loc)
 
-    case Expr.TryWith(_, _, _, _, tpe, _, _) =>
-      // TODO: VERIFIER: Add support for TryWith.
-      tpe
+    case Expr.TryWith(exp, effUse, rules, ct, tpe, purity, loc) =>
+      val exptype = visitExpr(exp) match {
+        case MonoType.Arrow(List(u), t) =>
+          check(expected = MonoType.Unit)(actual = u, loc)
```
[the remark this answers is not in the document]
Comment (felix-berg) [reply main/src/ca/uwaterloo/flix/language/phase/Verifier.scala:449]: this can probably be improved in other places too...

**Review by magnus-madsen — APPROVED** [review]

**Pull request merged**